src/octoprint/util/comm.py

```diff
@@ -949,10 +952,14 @@ def _getGcodeScript(self, scriptName, replacements=None):
 					continue
 
 				def to_list(data):
-					if isinstance(data, str):
-						data = map(str.strip, data.split("\n"))
-					elif isinstance(data, unicode):
-						data = map(unicode.strip, data.split("\n"))
+					if PY3:
```

Comment:
Come to think of it, wouldn't something like 

``` python
data = map(lambda x: x.strip(), data.split("\n"))
```

actually be the trivial solution here?

Reply:
Sorry, I don't get how you will get rid of the tests, and avoiding to have unicode reference when running python3 with that.

Reply:
What I mean is that this whole section of code with the `if`s is only there to decide what function to hand to `map` based on the type of `data`, because handing the wrong type to either function will result in an error: `str.strip` is it's a py2 str, `unicode.strip` if it's a py2 unicode.

But instead of figuring out what data type it is (which gets more complicated once py2/py3 need to coexist), we can just as well just say that we don't care as long as it has a `strip` function (which is an assumption here) and just go for that by calling it directly on the object in the `lambda` function. Probably wrapped with something like `if isinstance(data, basestring)` though to take care of the assumption.

I might be bad today at explaining myself for some reason...

Reply:
Ok, now I see it. I've always sucked understanding lambda things, don't like it